src/app/index.js

```diff
@@ -7,29 +9,68 @@ angular.module('app', [
   // 'angularMoment',
   // 'ngAnimate',
   // 'ui.bootstrap',
+  angularCookies,
+  satellizer,
   uiRouter
   // 'vModal'
 ]).config([
   '$stateProvider',
   '$urlRouterProvider',
   '$locationProvider',
-  ($stateProvider, $urlRouterProvider, $locationProvider) => {
+  '$authProvider',
+  ($stateProvider, $urlRouterProvider, $locationProvider, $authProvider) => {
     // Allow the use of HTML5 state modes
     $locationProvider.html5Mode(true)
 
+    let loginRequired = ['$q', '$location', '$auth', ($q, $location, $auth) => {
+      let deferred = $q.defer()
+      if ($auth.isAuthenticated()) {
+        deferred.resolve()
+      } else {
+        $location.path('/login')
+      }
+      return deferred.promise
+    }]
+
     // Define all of the application states
     $stateProvider
       .state('home', {
         url: '/',
         template: require('./components/home/homeView.html'),
         controller: 'HomeController',
-        controllerAs: 'vm'
+        controllerAs: 'homeCtrl',
+        onEnter: ['$state', '$auth', ($state, $auth) => {
+          if ($auth.isAuthenticated()) {
+            $state.go('dashboard')
+          }
+        }]
+      })
+      .state('dashboard', {
+        url: '/dashboard',
+        template: require('./components/dashboard/dashboardView.html'),
+        controller: 'DashboardController',
+        controllerAs: 'dashCtrl',
+        resolve: {
+          loginRequired: loginRequired
+        }
       })
       .state('login', {
         url: '/login',
         template: require('./components/login/loginView.html'),
         controller: 'LoginController',
-        controllerAs: 'vm'
+        controllerAs: 'loginCtrl',
+        onEnter: ['$state', '$auth', ($state, $auth) => {
+          if ($auth.isAuthenticated()) {
+            $state.go('dashboard')
+          }
+        }]
+      })
+      .state('logout', {
+        url: '/logout',
+        onEnter: ['$state', '$auth', ($state, $auth) => {
+          $auth.logout()
+          $state.go('login')
```

Comment:
I considered that choice, but ultimately I felt it made more sense to route them to login. Most websites that I use do that.